plugins/modules/adguardhome_rewrite.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025 Markus Bergholz
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+module: adguardhome_rewrite
+
+short_description: Add, update or delete DNS rewrite rules from AdGuard Home
+
+version_added: 3.3.0
+
+description:
+  - Add, update or delete DNS rewrite rules from AdGuard Home.
+extends_documentation_fragment:
+  - community.dns.adguardhome.connectivity
+options:
+  state:
+    description:
+      - wether a rewrite rule should be added/updated O(state=present) or removed O(state=absent).
+    type: str
+    default: present
+    choices:
+      - present
+      - absent
+  domain:
+    description:
+      - domain or wildcard domain that you want to be rewritten by AdGuard Home.
+    type: str
+    required: true
+  answer:
+    description:
+      - value for the domain rewrite.
+      - required when O(state=present).
+      - value can be a CNAME, A or AAAA record.
+    type: str
+    required: false
+attributes:
+  check_mode:
+    support: full
+    description: Can run in C(check_mode) and return changed status prediction without modifying target.
+    details:
+      - This action does not modify state.
+  diff_mode:
+    support: full
+    description: Will return details on what has changed (or possibly needs changing in C(check_mode)), when in diff mode.
+  idempotent:
+    support: full
+    description:
+      - When run twice in a row outside check mode, with the same arguments, the second invocation indicates no change.
+      - This assumes that the system controlled/queried by the module has not changed in a relevant way.
+
+
+author:
+  - Markus Bergholz (@markuman) <markuman+spambelongstogoogle@gmail.com>
+"""
+
+EXAMPLES = r"""
+- name: add dns rewrite rule in adguard home
+  community.dns.adguardhome_rewrite:
+    state: present
+    answer: 127.0.0.1
+    domain: example.org
+
+# when removing a rewrite, the current answer value must also match
+# therefore you can just leave it out and the existing value
+# will be used
+- name: remove rewrite for example.org
+  community.dns.adguardhome_rewrite:
+    state: absent
+    domain: example.org
+"""
+
+RETURN = r"""
+rules:
+  description: The list of fetched rewrite rules.
+  type: list
+  elements: dict
+  returned: always
```

Comment:
I guess this won't be returned if the module fails?
```suggestion
  returned: success
```